### ISSUE TITLE
Reporting coverage

### ISSUE THREAD

**Issue by fcooper8472** [rajeshrinet/pystokes#10]
I'm not sure how much effort would be required for this, but it would be great to automatically generate a code coverage report so you could see what proportion of the code is currently covered by tests.

If this was a pure Python library, it would be very straightforward using [coverage](https://coverage.readthedocs.io/en/coverage-5.1/) to generate the coverage info and [codecov](https://codecov.io/) to produce reports.

However, because this is a Cython library, I am not sure what would be required to get coverage reporting up-and-running, so for now this is only open as a suggestion.

**Comment by rajeshrinet** [comment]
@fcooper8472 thanks for the suggestion. I will read through their documentation and try to get it done for PyStokes. 

I must admit that the current testing is very minimal.

**Comment by rajeshrinet** [comment]
@fcooper8472 I have now added [![codecov](https://codecov.io/gh/rajeshrinet/pystokes/branch/master/graph/badge.svg)](https://codecov.io/gh/rajeshrinet/pystokes). Will add more testing!

**Comment by rajeshrinet** [comment]
in continuation, as you have noted it only tests .py files. I will see if it can be extended to check .pyx files as well.

**Comment by rajeshrinet** [comment]
@fcooper8472 as you pointed at the start of this issue, coverage in Cython codebase is a bit tricky. I did an example here: https://github.com/rajeshrinet/codecov-cython. 

I do not like ways in which I had to change the setup.py and force install the codebase for line coverage. It also leads to performance loss as explained here: http://blog.behnel.de/posts/coverage-analysis-for-cython-modules.html

Thus, I have decided to not implement it for PyStokes as I would like to squeeze every bit of performance possible. But, I would add a more complete unit tests gradually.
